src/coreclr/jit/importercalls.cpp

```diff
@@ -32,6 +32,107 @@
+    param.fncHandle       = methHnd;
+    param.exactContextHnd = MAKE_METHODCONTEXT(methHnd);
+    param.success         = false;
+
+    bool success = eeRunWithErrorTrap<Param>(
+        [](Param* pParam) {
+            // Cache some frequently accessed state.
+            //
+            Compiler* const       compiler    = pParam->pThis;
+            COMP_HANDLE           compCompHnd = compiler->info.compCompHnd;
+            CORINFO_METHOD_HANDLE ftn         = pParam->fncHandle;
+
+#ifdef DEBUG
+            if (JitConfig.JitNoInline())
+            {
+                pParam->success = false;
+                return;
+            }
+#endif
+
+            // Fetch method info. This may fail, if the method doesn't have IL.
+            //
+            CORINFO_METHOD_INFO methInfo;
+            if (!compCompHnd->getMethodInfo(pParam->fncHandle, &methInfo, pParam->exactContextHnd))
+            {
+                return;
+            }
+
+            // Speculatively check if initClass() can be done.
+            // If it can be done, we will try to inline the method.
+            CorInfoInitClassResult const initClassResult =
+                compCompHnd->initClass(nullptr /* field */, ftn /* method */, pParam->exactContextHnd /* context */);
+
+            if (initClassResult & CORINFO_INITCLASS_DONT_INLINE)
+            {
+                return;
+            }
+
+            // Given the VM the final say in whether to inline or not.
+            // This should be last since for verifiable code, this can be expensive
+            //
+            CorInfoInline const vmResult = compCompHnd->canInline(compiler->info.compMethodHnd, ftn);
+            if (vmResult == INLINE_FAIL || vmResult == INLINE_NEVER)
+            {
+                return;
+            }
+
+            pParam->resolvedToken.tokenContext = pParam->exactContextHnd;
+            pParam->resolvedToken.tokenScope   = methInfo.scope;
+            pParam->resolvedToken.tokenType    = CORINFO_TOKENKIND_Field;
+
+            auto code = methInfo.ILCode;
+            auto size = methInfo.ILCodeSize;
+            // instance getter
+            if (size == 7 && code[0] == CEE_LDARG_0 && code[1] == CEE_LDFLD && code[6] == CEE_RET)
+            {
+                pParam->resolvedToken.token = getU4LittleEndian(&code[2]);
+                compCompHnd->resolveToken(&pParam->resolvedToken);
+                pParam->opcode  = CEE_LDFLD;
+                pParam->success = true;
+                return;
+            }
+            // instance setter
+            else if (size == 8 && code[0] == CEE_LDARG_0 && code[1] == CEE_LDARG_1 && code[2] == CEE_STFLD &&
+                     code[7] == CEE_RET)
+            {
+                pParam->resolvedToken.token = getU4LittleEndian(&code[3]);
+                compCompHnd->resolveToken(&pParam->resolvedToken);
+                pParam->opcode  = CEE_STFLD;
+                pParam->success = true;
+                return;
+            }
```

Comment:
Maybe also do it for static properties?

Reply:
Originally I did but static's are more complicated and might need helper calls which bloat the callee and ran into "not supported" scenarios in the JIT so I've excluded them